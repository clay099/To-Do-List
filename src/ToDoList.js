import React, { useState } from "react";
import NewToDoForm from "./NewToDoForm";
import ToDo from "./ToDo";

const ToDoList = () => {
	const [toDoList, setToDoList] = useState([]);

	console.log("general todolist", toDoList);
	const addToDo = (newToDo) => {
		setToDoList((toDoList) => [...toDoList, newToDo]);
	};

	const deleteToDo = (task) => {
		setToDoList((listOfTodos) => {
			return listOfTodos.filter((todo) => todo.task !== task);
		});
	};

	return (
		<>
			<h1>To Do List</h1>
			<NewToDoForm addToDo={addToDo} />
			<div>
				{toDoList.map(({ task }) => {
					return <ToDo key={task} task={task} deleteToDo={deleteToDo} />;
				})}
			</div>
		</>
	);
};

export default ToDoList;
