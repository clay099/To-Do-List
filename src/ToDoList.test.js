import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import ToDoList from "./ToDoList";

beforeEach(() => {
	// due to adding localstorage, need to clear this otherwise tasks will keep being added
	localStorage.clear();
});
//smoke test
it("should render NewBoxForm", () => {
	render(<ToDoList />);
});

//snapshot test
it("should match snapshot", () => {
	const { asFragment } = render(<ToDoList />);
	expect(asFragment()).toMatchSnapshot();
});

function addTask(toDoList, taskName) {
	const todo = toDoList.getByLabelText("New Task");
	const btn = toDoList.getByText("Submit");

	fireEvent.change(todo, { target: { value: taskName } });
	fireEvent.click(btn);
}

it("should add a new todo", () => {
	const list = render(<ToDoList />);
	expect(list.queryByText("task")).not.toBeInTheDocument();

	addTask(list, "task");

	expect(list.queryByText("task")).toBeInTheDocument();
});

it("should delete a new todo", () => {
	const list = render(<ToDoList />);
	addTask(list, "new task");

	expect(list.queryByText("new task")).toBeInTheDocument();

	const del = list.getByText("Delete");
	fireEvent.click(del);

	expect(list.queryByText("new task")).not.toBeInTheDocument();
});

it("should edit a new todo", () => {
	const list = render(<ToDoList />);
	addTask(list, "new task");

	expect(list.queryByText("new task")).toBeInTheDocument();

	const edit = list.getByText("Edit");
	fireEvent.click(edit);

	fireEvent.change(list.getByDisplayValue("new task"), { target: { value: "updated" } });

	fireEvent.click(list.getByText("Update!"));

	expect(list.queryByText("updated")).toBeInTheDocument();
});
