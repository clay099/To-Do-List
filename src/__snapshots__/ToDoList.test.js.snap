// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <h1>
    To Do List
  </h1>
  <form
    class="NewTaskForm"
  >
    <label
      for="task"
    >
      New Task
    </label>
    <input
      id="task"
      name="task"
      placeholder="New Task"
      type="text"
      value=""
    />
    <button
      class="NewTaskForm-btn"
    >
      Submit
    </button>
  </form>
  <div />
</DocumentFragment>
`;
